apps/api/v2/src/modules/slots/slots-2024-09-04/slots.repository.ts

```diff
@@ -56,6 +56,29 @@ export class SlotsRepository_2024_09_04 {
     });
   }
 
+  async getExistingSlotsReservationCount(eventTypeId: number, startDate: string, endDate: string) {
+    return this.dbRead.prisma.selectedSlots.count({
+      where: {
+        eventTypeId,
+        slotUtcStartDate: startDate,
+        slotUtcEndDate: endDate,
+        // Only consider non-expired reservations
+        releaseAt: { gt: DateTime.utc().toJSDate() },
+      },
+    });
+  }
+
+  async getExistingBooking(eventTypeId: number, slotStartTime: Date, slotEndTime: Date) {
```

Comment:
This should be in `apps/api/v2/src/ee/bookings/2024-08-13/bookings.repository.ts` because it accesses `booking` model. Each repository should access only the table it is repository for aka booking repository only accesses bookings table, event type repository only event types table etc